bindings/python/tests/fdbcli_tests.py

```diff
@@ -82,6 +93,217 @@ def maintenance(logger):
+    class_type_line_2 = output2.split('\n')[-1]
+    logger.debug(class_type_line_2)
+    # check process' network address
+    assert '127.0.0.1' in class_type_line_2
+    # check class type changed to the specified value
+    assert random_class_type in class_type_line_2
+    # check class source
+    assert 'set_class' in class_type_line_2
+    # set back to default
+    run_fdbcli_command('setclass', '127.0.0.1', 'default')
+    # everything should be back to the same as before
+    output3 = run_fdbcli_command('setclass')
+    class_type_line_3 = output3.split('\n')[-1]
+    logger.debug(class_type_line_3)
+    assert class_type_line_3 == class_type_line_1
+
+@enable_logging()
+def lockAndUnlock(logger):
+    # lock an unlocked database, should be successful
+    output1 = run_fdbcli_command('lock')
+    # UID is 32 bytes
+    lines = output1.split('\n')
+    lock_uid = lines[0][-32:]
+    assert lines[1] == 'Database locked.'
+    logger.debug("UID: {}".format(lock_uid))
+    assert get_value_from_status_json(True, 'cluster', 'database_lock_state', 'locked')
+    # lock a locked database, should get the error code 1038
+    output2 = run_fdbcli_command_and_get_error("lock")
+    assert output2 == 'ERROR: Database is locked (1038)'
+    # unlock the database
+    process = subprocess.Popen(command_template + ['unlock ' + lock_uid], stdin = subprocess.PIPE, stdout = subprocess.PIPE)
+    line1 = process.stdout.readline()
+    # The randome passphrease we need to confirm to proceed the unlocking
+    line2 = process.stdout.readline()
+    logger.debug("Random passphrase: {}".format(line2))
+    output3, err = process.communicate(input=line2)
+    # No error and unlock was successful
+    assert err is None
+    assert output3.decode('utf-8').strip() == 'Database unlocked.'
+    assert not get_value_from_status_json(True, 'cluster', 'database_lock_state', 'locked')
+
+@enable_logging()
+def kill(logger):
+    output1 = run_fdbcli_command('kill')
+    lines = output1.split('\n')
+    assert len(lines) == 2
+    assert lines[1].startswith('127.0.0.1:')
+    address = lines[1]
+    logger.debug("Address: {}".format(address))
+    old_generation = get_value_from_status_json(False, 'cluster', 'generation')
+    # This is currently an issue with fdbcli, 
+    # where you need to first run 'kill' to initialize processes' list 
+    # and then specify the certain process to kill
+    process = subprocess.Popen(command_template[:-1], stdin = subprocess.PIPE, stdout = subprocess.PIPE)
+    # 
+    output2, err = process.communicate(input='kill; kill {}\n'.format(address).encode())
+    logger.debug(output2)
+    # wait for a second for the cluster recovery
+    time.sleep(1)
+    new_generation = get_value_from_status_json(True, 'cluster', 'generation')
+    logger.debug("Old: {}, New: {}".format(old_generation, new_generation))
+    assert new_generation > old_generation
+
+@enable_logging()
+def suspend(logger):
```

Comment:
I think this is potentially flaky. What do you think of this?

1. Get PID of fdbserver process
2. Suspend fdbserver process for a long time
3. Check that PID still exists and cluster is unavailable
4. kill PID (not using fdbcli, but by sending the PID a kill signal)
5. Confirm database becomes available

Reply:
Done. thanks